src/schema/firefox-schemas-import.js

```diff
@@ -353,8 +355,29 @@ export function rewriteExtend(schemas, schemaId) {
     (extendSchema.types || []).forEach((type) => {
       const { $extend, id, ...rest } = type;
       if ($extend) {
+        // Throw an explicit error if we are going to deep merge a set of properties which may be
+        // overriding an existing one, instead of extending the type with new ones as expected.
+        if (rest.properties) {
+          const newProps = Object.keys(rest.properties);
+          if (
+            !newProps.every(
+              (k) => extendDefinitions?.[$extend]?.properties?.[k] == null
```

Comment:
```suggestion
              (k) => extendDefinitions?.[$extend]?.properties?.[k] === null
```

?

Reply:
`== null` is true for both `null` and `undefined`, that's why I opted for `==` instead of `===` here

Reply:
I wasn't sure this was expected tbh so maybe add a comment?

Reply:
oh, yeah, that can be reasonable to remind ourselves that.

Reply:
comment added in one last patch added to this PR.